test/xds/xds_client_ignore_resource_deletion_test.go

```diff
@@ -0,0 +1,474 @@
+	}
+	t.Cleanup(func() { cc.Close() })
+
+	verifyRPCtoAllEndpoints(t, cc)
+
+	// Mutate resource and update on the server.
+	u(&resources)
+	if err := ms.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Spin up go routines to verify RPCs fail after the update.
+	client := testServicepb.NewTestServiceClient(cc)
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		for ctx.Err() == nil {
+			if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+				wg.Done()
+				break
+			}
+			time.Sleep(50 * time.Millisecond)
+		}
+	}()
+	go func() {
+		for ctx.Err() == nil {
+			if _, err := client.UnaryCall(ctx, &testpb.SimpleRequest{}); err != nil {
+				wg.Done()
+				break
+			}
+			time.Sleep(50 * time.Millisecond)
+		}
+	}()
+
+	wg.Wait()
+	if ctx.Err() != nil {
+		t.Fatal("Context expired before RPCs failed.")
+	}
+}
+
+// This helper creates a management server for the test.
+func startManagementServer(t *testing.T) *e2e.ManagementServer {
+	t.Helper()
+	ms, err := e2e.StartManagementServer(e2e.ManagementServerOptions{})
+	if err != nil {
+		t.Fatalf("Failed to start management server: %v", err)
+	}
+	t.Cleanup(ms.Stop)
+	return ms
+}
+
+// This helper generates a custom bootstrap config for the test.
+func generateBootstrapContents(t *testing.T, serverURI string, ignoreResourceDeletion bool) []byte {
+	t.Helper()
+	bootstrapContents, err := bootstrap.Contents(bootstrap.Options{
+		NodeID:                             nodeID,
+		ServerURI:                          serverURI,
+		ServerListenerResourceNameTemplate: e2e.ServerListenerResourceNameTemplate,
+		IgnoreResourceDeletion:             ignoreResourceDeletion,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	return bootstrapContents
+}
+
+// This helper generates a custom bootstrap config for the test.
+func generateCustomResolver(t *testing.T, bootstrapContents []byte) resolver.Builder {
+	cResolver, err := resolverBuilder(bootstrapContents)
+	if err != nil {
+		t.Fatalf("Creating xDS resolver for testing: %v", err)
+	}
+	return cResolver
+}
+
+// This helper creates an xDS-enabled gRPC server using the listener and the
+// bootstrap config passed. This helper then registers the server to testService
+// and returns a func to accept requests.
+func setupGRPCServerWithModeChangeChannel(t *testing.T, bootstrapContents []byte, lis net.Listener) (chan connectivity.ServingMode, func()) {
+	t.Helper()
+	updateCh := make(chan connectivity.ServingMode, 1)
+
+	// Create a server option to get notified about serving mode changes.
+	modeChangeOpt := xds.ServingModeCallback(func(addr net.Addr, args xds.ServingModeChangeArgs) {
+		t.Logf("serving mode for listener %q changed to %q, err: %v", addr.String(), args.Mode, args.Err)
+		updateCh <- args.Mode
+	})
+
+	server := xds.NewGRPCServer(grpc.Creds(insecure.NewCredentials()), modeChangeOpt, xds.BootstrapContentsForTesting(bootstrapContents))
+	t.Cleanup(server.Stop)
+	testpb.RegisterTestServiceServer(server, &testService{})
```

Comment:
This and other places where you use symbols from the generated code which refer to services (instead of messages) should use `testgrpc`.

Reply:
done